internal/syncd/integrations/github/listener_test.go

```diff
@@ -0,0 +1,97 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+type mockDownloader struct {
+	mock.Mock
+}
+
+func (m *mockDownloader) BufferDirectory(ctx context.Context, repo, path, ref string) ([]*chartutil.BufferedFile, error) {
+	args := m.Called(ctx, repo, path, ref)
+	var ret0 []*chartutil.BufferedFile
+	if args0 := args.Get(0); args0 != nil {
+		ret0 = args0.([]*chartutil.BufferedFile)
+	}
+
+	return ret0, args.Error(1)
+}
+
+type mockReconciler struct {
+	mock.Mock
+}
+
+func (m *mockReconciler) Reconcile(ctx context.Context, chart *chart.Chart) error {
+	args := m.Called(ctx, chart)
+	return args.Error(0)
+}
+
+func TestHandlerChartDownloadFails(t *testing.T) {
+	errNotFound := errors.New("not found")
+	testEvent := pushEvent{
+		Ref:        "ref",
+		Path:       "path",
+		Repository: "repository",
+	}
+
+	eventBytes, err := json.Marshal(testEvent)
+	if !assert.NoError(t, err) {
```

Comment:
Is this `if` statement necessary? Would the test case fail in the case of an error by just calling `assert.NoError` without explicitly telling it do so through an `if` statement? This pattern is repeated elsewhere so perhaps I am missing something.

Reply:
Wondering this too.

Reply:
`assert.NoError` causes the test case to fail, but doesn't stop the test case execution. If I don't check the return value and call `t.FailNow()`, the test case will continue on and panic or produce a meaningless result because `eventBytes === nil`.